src/profile/hooks/useProfileDataEditor.ts

```diff
@@ -1,34 +1,109 @@
-import { useContext, useMemo } from 'react';
+import { useEffect, useMemo, useState } from 'react';
+import to from 'await-to-js';
 
-import { ProfileContext } from '../context/ProfileContext';
 import {
   EditData,
   EditDataType,
   createEditorForDataType,
+  SaveType,
+  EditDataValue,
+  FormValues,
 } from '../helpers/editData';
+import { UpdateResult, useProfileMutations } from './useProfileMutations';
 
 export type ProfileDataEditorReturnType = {
   editDataList: EditData[];
+  save: (item: EditData, newValue: EditDataValue) => Promise<UpdateResult>;
+  reset: (item: EditData) => void;
 };
 
+export type Action = SaveType | 'edit' | 'cancel' | 'save' | 'add';
```

Comment:
What's the difference between SaveType and 'save'? Sounds semantically similar.

Reply:
SaveType in EditData tells what property is being saved: 'values' or 'set-primary'.  'save' is just an general action.